### PR TITLE
VueJS TODO

### PR DESCRIPTION
**Pull request by dreyhiflden**

### CODE REVIEW

File: src/components/AddTaskForm.vue [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  </div>
+</template>
+
+<script>
+  export default {
+    data: function() {
+      return {
+        newTodoTitle: null,
+        newTodoDescription: null,
+      }
```
Comment (carrot-e) [inline src/components/AddTaskForm.vue]: `;`

File: src/components/AddTaskForm.vue [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    },
+    methods: {
+      addItem () {
```
Comment (carrot-e) [inline src/components/AddTaskForm.vue]: `addItem()`

File: src/components/AddTaskForm.vue [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        let inputData = {
```
Comment (carrot-e) [inline src/components/AddTaskForm.vue]: `inputData` is used only once.

File: src/components/TodoList.vue [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        this.todos.push({
+          id: nextTodoId++,
+          text: inputData.text,
+          description: inputData.description,
+        });
+      },
+      removeTodo (idToRemove) {
+        this.todos = this.todos.filter(todo => {
+          return todo.id !== idToRemove
+        })
```
Comment (carrot-e) [inline src/components/TodoList.vue]: `;`

File: src/components/TodoList.vue [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  .empty-list {
+    text-align: center;
+  }
+
+  .list-enter-active, .list-leave-active {
+    transition: all 0.3s;
+  }
+  .list-enter, .list-leave-to /* .list-leave-active до версии 2.1.8 */ {
```
Comment (carrot-e) [inline src/components/TodoList.vue]: ?

File: src/components/TodoListItem.vue [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+      },
+    },
+    data: function() {
+      return {
+        isHovered: false,
+        isEditing: false,
+      }
+    },
+    methods: {
+      blur(e) {
```
Comment (carrot-e) [inline src/components/TodoListItem.vue]: :sob: 
You are manipulating the DOM which you should never do in components.

File: src/components/AddTaskForm.vue (hunk 1 above)
Comment (dreyhiflden) [reply src/components/AddTaskForm.vue]: When I put `;` instead `,` I have error while compiling project.

Comment (carrot-e) [reply src/components/AddTaskForm.vue]: Of course, you get an error.
The semicolon should appear in line 29.

File: src/components/TodoListItem.vue (hunk 6 above)
Comment (dreyhiflden) [reply src/components/TodoListItem.vue]: In this case I must use vanilla JS for this?

File: src/components/AddTaskForm.vue (hunk 1 above)
Comment (dreyhiflden) [reply src/components/AddTaskForm.vue]: And you was totally right :)

File: src/components/TodoListItem.vue (hunk 6 above)
Comment (dreyhiflden) [reply src/components/TodoListItem.vue]: Fixed

File: src/components/TodoList.vue [hunk 7]
```diff
@@ -0,0 +1,92 @@
+<template>
+  <div>
+    <AddTaskForm @addItem="addTodo" />
+    <ul v-if="todos.length" class="todo-list">
+      <transition-group name="list" tag="div" id="todolist">
+        <TodoListItem
+          v-for="todo in todos"
+          :key="todo.id"
+          :todo="todo"
+          :ispressed="ispressed"
```
Comment (carrot-e) [inline src/components/TodoList.vue]: `is-pressed` is more readable, isn't it?
And the name isn't quite clear. What `is pressed`? How `is pressed` should affect the UI?

File: src/components/TodoListItem.vue [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+    },
+    data: function() {
+      return {
+        isHovered: false,
+        isEditing: false,
+      }
+    },
+    methods: {
+      deleteItem(todoid) {
+        this.$emit('remove', todoid);
```
Comment (carrot-e) [inline src/components/TodoListItem.vue]: `todoId`

File: src/components/TodoListItem.vue [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+    width: 118px;
+    outline: none;
+  }
+
+  .hover-details__button:active {
+    background: rgba(255, 255, 255, 0.05);
+  }
+
+  .hover-details__button--green {
+    color: #9DC876;
```
Comment (carrot-e) [inline src/components/TodoListItem.vue]: Lowercase?

Comment (dreyhiflden) [reply src/components/TodoListItem.vue]: Change to lowercase

File: src/components/TodoListItem.vue (hunk 8 above)
Comment (dreyhiflden) [reply src/components/TodoListItem.vue]: Change to `todoId`

File: src/components/TodoList.vue (hunk 7 above)
Comment (dreyhiflden) [reply src/components/TodoList.vue]: Changed to `isEditMode`

**Comment by carrot-e** [comment]
Unit tests are still to be done though :wink:

**Pull request merged**